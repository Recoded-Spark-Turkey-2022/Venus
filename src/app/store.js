// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
