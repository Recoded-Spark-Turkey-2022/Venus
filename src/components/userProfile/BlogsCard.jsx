import React from "react";

const BlogsCard = ({title,name,image,text})=> {

    return(
        <div className="bg-white max-w-sm flex justify-center rounded-3xl overflow-hidden shadow-md py-8 px-3 mb-2">
            <div className="text-xl mb-2 ">
                <div className="py-2">
            <p className="font-bold">{title}</p>
            </div>
            <div className="py-2">
            <p>{text}</p>
            </div>
            <div className="flex justify-start">
            <div className="max-w-[25%] bg-orange-500 justify-center items-center rounded-full">
                <img className="rounded-full" src={image} alt="Avatar of User" />
            </div>
                <p className="text-md text-mediumBlue pt-10 pl-5">{name}</p>
                </div>
    
          </div>
        </div>

    )
}
export default BlogsCard;