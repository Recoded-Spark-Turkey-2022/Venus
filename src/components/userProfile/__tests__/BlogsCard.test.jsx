import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../../app/store';
import BlogsCard from '../BlogsCard';


describe('BlogsCard', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Router>
            <Provider store={store}>
              <BlogsCard />
            </Provider>
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });