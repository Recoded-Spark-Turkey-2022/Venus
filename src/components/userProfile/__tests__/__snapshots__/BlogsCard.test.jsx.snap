// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogsCard renders correctly 1`] = `
<div
  className="bg-white w-[100%] md:w-[70%] lg:w-[400px] flex h-[300px]  justify-center rounded-3xl overflow-hidden shadow-md py-8 px-3 mb-2"
>
  <div
    className="text-xl mb-2 text-left  flex flex-col justify-between"
  >
    <div
      className="py-2"
    >
      <p
        className="font-bold blog-text"
      />
    </div>
    <div
      className="py-2"
    >
      <p
        className="blog-text"
      />
    </div>
    <div
      className="flex justify-start"
    >
      <div
        className="w-[70px]   rounded-full"
      >
        <img
          alt="Avatar of User"
          className="rounded-full  aspect-square"
          src="https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg"
        />
      </div>
      <p
        className="text-md text-mediumBlue pt-10 pl-5"
      />
    </div>
  </div>
</div>
`;
